### PR TITLE
Initialise Shared::DAMAGE in PrognosticData::update

### LINKED ISSUE #742: Incorrect initialisation of damage
Damage is incorrectly initialised (at least in the benchmark case). Outputting every time step, damage after the first time step looks like this:

image

However, there should be no slope towards zero going towards x=0. Turning off the wind makes the problem more obvious:

image

### PR DESCRIPTION
**Pull request by einola**
#  Initialise Shared::DAMAGE in PrognosticData::update

Fixes #742 

### Task List
- [x] Defined the tests that specify a complete and functioning change (*It may help to create a [design specification & test specification](../../../wiki/Specification-Template)*)
- [x] Implemented the source code change that satisfies the tests
- [x] Documented the feature by providing worked example
- [x] Updated the README or other documentation
- [x] Completed the pre-Request checklist below

---
# Change Description

The ``Shared::DAMAGE`` variable must be updated at the start of ``PrognosticData::update``, otherwise the ``kernel.setData(damageName, damage)`` call in ``BBMDynamics.cpp`` (line 123) inserts nonsense into the damage field.

---
# Test Description

Use a modified config_benchmark.cfg file to test the update

```
[model]
init_file = init_benchmark_128x128.nc
start = 2023-01-01T00:00:00Z
stop = 2023-01-01T00:10:00Z
time_step = P0-0T00:02:00

[Modules]
DiagnosticOutputModule = Nextsim::ConfigOutput
DynamicsModule = Nextsim::BBMDynamics
IceThermodynamicsModule = Nextsim::DummyIceThermodynamics
LateralIceSpreadModule = Nextsim::DummyIceSpread
AtmosphereBoundaryModule = Nextsim::ConfiguredAtmosphere
OceanBoundaryModule = Nextsim::BenchmarkOcean

[ConfigOutput]
period = P0-0T00:02:00
start = 2010-01-01T12:00:00Z
field_names = hsnow,hice,cice,u,v,damage
filename = benchmark_32x32.diagnostic.nc
```

Before the fix, the damage field would have a uniform slope from x=0 to half way through the domain. After the fix, the damage field is uniform as it should be.

---
# Documentation Impact

N/A

---
# Other Details

N/A

---
### Pre-Request Checklist

- [x] The requirements of this pull request are fully captured in an issue or design specification and are linked and summarised in the description of this PR
- [x] No new warnings are generated
- [x] The documentation has been updated (or an issue has been created to track the corresponding change)
- [x] Methods and Tests are commented such that they can be understood without having to obtain additional context
- [x] This PR/Issue is labelled as a bug/feature/enhancement/breaking change
- [x] File dates have been updated to reflect modification date
- [x] This change conforms to the conventions described in the README

### CODE REVIEW

**Comment by timspainNERSC** [comment]
An alternative suggestion, so that `damage` works like all the other fields: 
In `ConstantHealing.hpp` add:
```
    ModelArrayRef<Protected::DAMAGE> oldDamage;
```
In `ConstantHealing.cpp` change
```
    damage[i] = (damage[i] * (cice[i] - lateralGrowth) + lateralGrowth) / cice[i];
```
to 
```
    damage[i] = (oldDamage[i] * (cice[i] - lateralGrowth) + lateralGrowth) / cice[i];
```
But still delete the reference to `ticeUpd` in `PrognosticData.cpp`.

**Comment by einola** [comment]
Yes, that makes more sense. It is at least more in the spirit of the thing. But then I also have to do ``damage = oldDamage`` in ``NoHealing``.

**Comment by einola** [comment]
I'll do that and then get to  #725 sometime later.

**Comment by einola** [comment]
With 

```
    ModelArrayRef<Shared::DAMAGE, RW> damage; 
    ModelArrayRef<Protected::DAMAGE, RW> oldDamage;
```

right? It crashes with a segfault/bad access, so I'm probably misunderstanding something :/

**Comment by timspainNERSC** [comment]
> ModelArrayRef<Protected::DAMAGE, RW> oldDamage;

The array at `Protected::DAMAGE` is shared read-only, so attempting to access it read-write causes an access violation. The error message could be a little more subtle, but that's how it currently is.
```
ModelArrayRef<Protected::DAMAGE, RO> oldDamage;
```

**Comment by einola** [comment]
Yes, I realised just after I posted :)

File: physics/src/modules/DamageHealingModule/include/NoHealing.hpp [hunk 1]
```diff
@@ -26,7 +26,14 @@ class NoHealing : public IDamageHealing {
     ModelState getStateRecursive(const OutputSpec& os) const override { return ModelState(); };
 
     // Do nothing when update is called.
-    void update(const TimestepTime& tstep) override {};
+    void update(const TimestepTime& tstep) override
```
Comment (timspainNERSC) [inline physics/src/modules/DamageHealingModule/include/NoHealing.hpp]: `damage.data() = oldDamage.data();` should work without the `overElements()` call.

**Review by timspainNERSC — APPROVED** [review]

**Pull request merged**